posix.go

```diff
@@ -458,11 +458,9 @@ func (s posix) AppendFile(volume, path string, buf []byte) (n int64, err error)
 		}
 		return 0, err
 	}
-	if err = checkDiskFree(s.diskPath, s.minFreeDisk); err != nil {
-		return 0, err
-	}
 	filePath := pathJoin(volumeDir, path)
-	if err = checkPathLength(filePath); err != nil {
+	err = checkPathLength(filePath)
+	if err != nil {
 		return 0, err
```

Comment:
how come you broke `if err = checkPathLength(filePath); err != nil {}` style to this style? does it give any advantage?